src/jit/regalloc.cpp

```diff
@@ -595,10 +595,12 @@ void                Compiler::raSetupArgMasks(RegState *regState)
             continue;
 
         // only process args that apply to the current register file
+#ifndef ARM_SOFTFP
         if ((argDsc->IsFloatRegType() && !info.compIsVarArgs) != regState->rsIsFloat)
```

Comment:
Instead of this #ifdef, I think the following line should read:

if ((argDsc->IsFloatRegType() && !info.compIsVarArgs && !opts.compUseSoftFP) != regState->rsIsFloat)